test/cljam/io/cram/encode/alignment_stats_test.clj

```diff
@@ -49,3 +49,19 @@
            (into {}
                  (stats/merge-stats [{:ri -1, :start 0, :end 0, :nbases 10000, :nrecords 20}
                                      {:ri -1, :start 0, :end 0, :nbases 15000, :nrecords 30}]))))))
+
+(deftest make-alignment-spans-builder-test
+  (let [builder (stats/make-alignment-spans-builder)
+        records [{:ri 0, :start 1, :end 100}
+                 {:ri 0, :start 51, :end 150}
+                 {:ri 1, :start 51, :end 150}
+                 {:ri 1, :start 151, :end 300}
+                 {:ri -1, :start 0, :end 0}
+                 {:ri -1, :start 0, :end 0}]]
+    (run! (fn [{:keys [ri start end]}]
+            (stats/update-span! builder ri start end))
+          records)
+    (is (= {0 {:start 1, :span 150}
+            1 {:start 51, :span 250}
+            -1 {:start 0, :span 1}}
+           (stats/build-spans builder)))))
```

Comment:
> I think it would be beneficial to include test data where there's a gap in the alignment range for the same reference sequence id.

Will do. Thanks!

> Additionally, I'm not confident about my understanding of the specification, but is it correct that gaps are ignored?

I believe so. Otherwise, an index entry would have to be created for each contiguous alignment span.
The CRAM specification says:

> Multi-reference slices may need to have multiple lines for the same slice; one for each reference contained within that slice.

In other words, at most one index entry can be created for each reference, which is incompatible with the "one entry for each contiguous alignment span" assumption.

Reply:
As for the first point, c2ea942 just added a test case.